code/modules/mob/dead/observer/observer_base.dm

```diff
@@ -375,6 +375,74 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		return
 	GLOB.ghost_hud_panel.ui_interact(src)
 
+/mob/dead/observer/verb/respawn_character()
+	set name = "Respawn as New Character"
+	set category = "Ghost"
+
+	if(!GLOB.configuration.general.respawn_enabled && !check_rights(R_ADMIN))
+		to_chat(usr, "<span class ='warning'>Respawning has been disabled.</span>")
+		return
+
+	if(stat != DEAD)
+		to_chat(usr, "<span class ='notice'>You must be dead to respawn!</span>")
+		return
+
+	if(!SSticker || SSticker.current_state < GAME_STATE_PLAYING)
+		to_chat(src, "<span class ='warning'>You can't respawn before the game starts!</span>")
+		return
+
+	if(GAMEMODE_IS_WIZARD || GAMEMODE_IS_NUCLEAR)
+		to_chat(usr, "<span class ='warning'>You can't respawn for this gamemode.</span>")
+		return
+
+	var/deathtime = world.time - timeofdeath
+	if(!HAS_TRAIT(src, TRAIT_RESPAWNABLE) && !check_rights(R_ADMIN))
+		to_chat(usr, "<span class ='warning'>You don't have respawnability!</span>")
+		return
+
+	var/deathtimeminutes = round(deathtime / 600)
```

Comment:
camel_case these variable names, please. 
make it `death_time_minutes`/`death_time`